src/lib/parameters/parameters.cpp

```diff
@@ -1404,11 +1451,43 @@ param_import_callback(bson_decoder_t decoder, void *priv, bson_node_t node)
 	return result;
 }
 
+static int
+param_dump_callback(bson_decoder_t decoder, void *priv, bson_node_t node)
+{
+	switch (node->type) {
+	case BSON_EOO:
+		PX4_INFO_RAW("BSON_EOO\n");
+		return 0;
+
+	case BSON_DOUBLE:
+		PX4_INFO_RAW("BSON_DOUBLE: %s = %.6f\n", node->name, node->d);
+		return 1;
+
+	case BSON_BOOL:
+		PX4_INFO_RAW("BSON_BOOL:   %s = %d\n", node->name, node->b);
+		return 1;
+
+	case BSON_INT32:
+		PX4_INFO_RAW("BSON_INT32:  %s = %" PRIi32 "\n", node->name, node->i32);
+		return 1;
+
+	case BSON_INT64:
+		PX4_INFO_RAW("BSON_INT64:  %s = %" PRIi64 "\n", node->name, node->i64);
+		return 1;
+
+	default:
+		PX4_INFO_RAW("ERROR %s unhandled bson type %d\n", node->name, node->type);
+		return 1; // just skip this entry
+	}
+
+	return -1;
+}
+
 static int
 param_import_internal(int fd, bool mark_saved)
 {
 	for (int attempt = 1; attempt < 5; attempt++) {
```

Comment:
Do we still need these attempts?

Reply:
> Do we still need these attempts?

Debatable, but right now you're pretty much screwed without parameters so it's worth trying anything.

I'm hoping these changes allow us to capture actual data from these rare failures and then we can re-evaluate.